scripts/entity/components/attack_components/base_attack_component.gd

```diff
@@ -1,11 +1,46 @@
 class_name BaseAttackComponent extends Node
 
+signal on_target_change(previous_target, current_target)
+
+## The Area2D that the character locks onto the closest target.
+@export var lock_on_area: Area2D
+
+## The node to compare against the targets.
+@export var measurement_position: Node2D
+
+
+## The current target that is closest in the area.
+var _current_target: Node2D = null
 
 # Called when the node enters the scene tree for the first time.
 func _ready() -> void:
-	pass # Replace with function body.
+	on_target_change.connect(target_changed)
```

Comment:
What's target_changed? I don't see it defined

Reply:
It is at the top

Reply:
![image](https://github.com/user-attachments/assets/00fc64f2-ddfa-4d80-b26e-c3f6c3305d80)
Where?

Reply:
line 3